sound/soc/sof/ipc4-topology.c

```diff
@@ -1384,7 +1384,9 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	int *ipc_config_size;
 	u32 **data;
 	int ipc_size, ret;
-	u32 out_ref_rate, out_ref_channels, out_ref_valid_bits;
+	u32 out_ref_rate = 0;
```

Comment:
in one way this fix will avoid the warning but I dont like the fact that we will pass 0 for the default case. In reality the types can only be one of the 5 in the switch case because for everything else we'd have returned an error on line 151. So a better fix is:

```
switch (swidget->id) {
	case snd_soc_dapm_aif_in:
	case snd_soc_dapm_dai_out:
	case snd_soc_dapm_buffer:
	{
		struct sof_ipc4_audio_format *in_fmt;

		in_fmt = &available_fmt->input_pin_fmts[ret].audio_fmt;
		out_ref_rate = in_fmt->sampling_frequency;
		out_ref_channels = SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(in_fmt->fmt_cfg);
		out_ref_valid_bits = SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(in_fmt->fmt_cfg);
		break;
	}
	default:
		out_ref_valid_bits = sof_ipc4_get_valid_bits(sdev, fe_params);
		if (out_ref_valid_bits < 0)
			return out_ref_valid_bits;

		out_ref_rate = params_rate(fe_params);
		out_ref_channels = params_channels(fe_params);
		break;
	}
```

Reply:
Ack, made sense.